File: tournaments/isDiagonalMatrix/isDiagonalMatrix.js
function isDiagonalMatrix(matrix) {
    const n = matrix.length;

    for (let i = 0; n > i; ++i)
        for (let j = 0; n > j; ++j)
            if (i !== j && matrix[i][j])
                return false;

    return true;
}

File: tournaments/isDiagonalMatrix/isDiagonalMatrix.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub isDiagonalMatrix {
    my ($matrix) = @_;

    my $n = @$matrix;

    for (my $i = 0; $n > $i; ++$i) {
        for (my $j = 0; $n > $j; ++$j) {
            if ($i != $j && @{@$matrix[$i]}[$j]) {
                return 0;
            }
        }
    }

    return 1;
}
